cloud/shared/bin/lib/terraform.py

```diff
@@ -0,0 +1,71 @@
+import subprocess
+import os
+import shutil
+
+from cloud.shared.bin.lib import civiform_mode
+
+
+def perform_apply():
+    '''Generates terraform variable files and runs terraform init and apply.'''
+
+    TERRAFORM_PLAN_OUT_FILE = "terraform_plan"
+    TERRAFORM_TEMPLATE_DIR = os.getenv("TERRAFORM_TEMPLATE_DIR")
+    TF_VARS_FILENAME = os.getenv('TF_VAR_FILENAME')
+
+    terraform_cmd = ['terraform', f'-chdir={TERRAFORM_TEMPLATE_DIR}']
+
+    if civiform_mode.is_dev():
+        subprocess.check_call(
+            terraform_cmd + ['init', '-upgrade', '-reconfigure'])
+    else:
```

Comment:
Was the call to setup_tf_shared_state that was in the original version now unnecessary?

If we're going to eventually use this for Azure deployments as well, I'd leave a TODO.

Reply:
That's the bucket that was azure specific. See todo on top.